python/mxnet/gluon/estimator/estimator.py

```diff
@@ -241,16 +240,35 @@ def fit(self, train_data,
             from a data batch and load into contexts(devices)
         """
 
-
         self.epochs = epochs
-        if not batch_size:
-            batch_size = 32 * len(self.context)
+        if isinstance(train_data, gluon.data.DataLoader):
+            num_batches = len(train_data)
+            total_samples = len(train_data._dataset)
+            if total_samples == 0:
+                raise ValueError("DataLoader is Empty. Please refer to gluon.data.DataLoader "
+                                 "for more detail")
```

Comment:
nit: for more detail on how to contruct a dataloader?

Reply:
Done. Thanks!